src/CommonITILObject.php

```diff
@@ -8924,6 +8928,42 @@ protected function updateActors(bool $disable_notifications = false)
                         }
                     }
                 }
+
+
+                if (array_key_exists($actors_id_replace_input_key, $this->input)) {
+                    // when input key used is _replace_$actor_$actor_type(provided by rule engine)
+                    // Ex: _replace_groups_assign    or _replace_users_assign    or _replace_suppliers_assign
+                    // Ex: _replace_groups_observer  or _replace_users_observer  or _replace_suppliers_observer
+                    // Ex: _replace_groups_requester or _replace_users_requester or _replace_suppliers_requester
+                    // this indicates a complete replacement of actors of the same type
+
+                    // remove all related actors
+                    $actors_deleted_input_key = sprintf('_%s_%s_deleted', $actor_fkey, $actor_type);
+                    foreach ($this->{"lazy_loaded_" . str_replace("_id", "", $actor_fkey)} as $current_itilobject_actor) {
+                        foreach ($current_itilobject_actor as $actor_value) {
+                            $this->input[$actors_deleted_input_key][] =
+                            [
+                                "itemtype" => $actor_itemtype,
+                                "id" => $actor_value["id"],
+                            ];
+                        }
+                    }
```

Comment:
1. You should use `users`/`groups`/`suppliers` properties, not the `lazy_loaded_xxx` properties directly.
2. You should clean only the current actor type (assign, requester or observer), not all the actors.
```suggestion
                    $existing_actors = $this->{str_replace("_id", "", $actor_fkey)}; // `$this->users` or `$this->groups` or `$this->suppliers`
                    foreach ($existing_actors[$actor_type_value] as $actor_value) {
                        $this->input[$actors_deleted_input_key][] = [
                            "itemtype" => $actor_itemtype,
                            "id" => $actor_value["id"],
                        ];
                    }
```

Reply:
apply